### PR TITLE
Fix voucher apply once per order

### LINKED ISSUE #4314: Voucher one per order doesn't work
The once per user option doesn't work on percentage discounts. It always gives percentage of the whole order value. But it works for dollar amounts.

### PR DESCRIPTION
**Pull request by fowczarek**
I want to merge this change because, resolve #4314

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for c24bfa53c45c1d233d11182b928a9c0a5587a38f (mirumee/saleor @ fix_voucher_one_per_order)
Base comparison is f7897c7a9b67086b2963153cb848b63d57120fde.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                           	left count 	right count
  ------------------------------------	-----------	-----------
  add billing address to checkout     	         41	         41
  add shipping to checkout            	          7	          7
  checkout payment charge             	         16	         16
  complete checkout                   	          6	          6
  create checkout                     	         45	         45

# api.benchmark homepage
  test name                           	left count 	right count
  ------------------------------------	-----------	-----------
  retrieve main menu                  	          5	          5
  retrieve product list               	          4	          4
  retrieve secondary menu             	          5	          5
  retrieve shop                       	          2	          2

# api.benchmark product
  test name                           	left count 	right count
  ------------------------------------	-----------	-----------
  product details                     	         16	         16

# api.benchmark variant
  test name                           	left count 	right count
  ------------------------------------	-----------	-----------
  retrieve variant list               	         13	         13
```

</p>
</details>

File: tests/api/test_discount.py [hunk 1]
```diff
@@ -3,11 +3,11 @@
 import graphene
 import pytest
 from django_countries import countries
-from tests.api.utils import get_graphql_content
 
 from saleor.discount import DiscountValueType, VoucherType
 from saleor.discount.models import Sale, Voucher
 from saleor.graphql.discount.enums import DiscountValueTypeEnum, VoucherTypeEnum
+from tests.api.utils import get_graphql_content
```
Comment (korycins) [inline tests/api/test_discount.py]: Relative

File: tests/api/test_discount.py [hunk 2]
```diff
@@ -271,28 +275,34 @@ def test_create_voucher_with_existing_voucher_code(
...
                 voucher {
                     code
                     discountValueType
+                    applyOncePerOrder
                 }
             }
         }
     """
     # Set discount value type to 'fixed' and change it in mutation
+    apply_once_per_order = not voucher.apply_once_per_order
```
Comment (korycins) [inline tests/api/test_discount.py]: You put this line between line's comment and line. You should keep them together or remove the comment if it is not needed

File: tests/conftest.py [hunk 3]
```diff
@@ -508,6 +519,12 @@ def product_list(product_type, category):
             ),
         ]
     )
+    sku = 0
+    for product in products:
+        ProductVariant.objects.create(
+            product=product, sku=sku, track_inventory=True, quantity=100
+        )
+        sku = sku + 1
```
Comment (korycins) [inline tests/conftest.py]: This could generate a problem with db because sku is unique

File: tests/conftest.py [hunk 4]
```diff
@@ -93,6 +93,17 @@ def checkout_with_item(checkout, product):
     return checkout
 
 
+@pytest.fixture
+def checkout_with_items(checkout, product_list, product):
```
Comment (korycins) [inline tests/conftest.py]: Isn't `checkout_with_item` enough for you? This fixture generates a huge amount of unneeded data and it is time and db expensive.

File: tests/test_checkout.py [hunk 5]
```diff
@@ -849,6 +849,38 @@ def test_get_discount_for_checkout_value_voucher(
...
+    discount_amount,
+):
+    voucher = Voucher(
+        code="unique",
+        type=VoucherType.PRODUCT,
+        discount_value_type=discount_type,
+        discount_value=discount_value,
+        apply_once_per_order=apply_once_per_order,
+    )
+    voucher.save()
```
Comment (korycins) [inline tests/test_checkout.py]: Why not `Voucher.objects.create`?

File: tests/conftest.py (hunk 3 above)
Comment (NyanKiyoshi) [reply tests/conftest.py]: Are you sure? We have something like `123` or `555` in other places.

Comment (korycins) [reply tests/conftest.py]: Try to use fixtures `product` and `unavailable_product_with_variant` in the same test ;)
If we don't use tools like faker we should keep in mind that unique fields shouldn't be duplicated (like `uuid4()`)

**Pull request merged**